rest_framework-stubs/exceptions.pyi

```diff
@@ -15,7 +16,7 @@ class ErrorDetail(str):
     code: str | None
     def __new__(cls, string: str, code: str | None = ...): ...
 
-_Detail: TypeAlias = str | list[Any] | dict[str, Any]
+_Detail: TypeAlias = str | _StrPromise | list[Any] | dict[str, Any]
```

Comment:
Don't import private types from django-stubs. Use `StrPromise` from `django_stubs_ext`.

Reply:
Although I notice we currently don't have an explicit dependency on django-stubs-ext in `setup.py`. IMO that should that be added, thoughts @sobolevn?

Reply:
Also this could use the `StrOrPromise` type alias:

```suggestion
_Detail: TypeAlias = StrOrPromise | list[Any] | dict[str, Any]
```

Reply:
`django-stubs-ext` is a dependency of `django-stubs`. I don't think we need to duplicate it 🤔

Reply:
Sorry to hijack this discussion but I can't find any specific documentation anywhere about this. Just want to confirm that when I have code that uses `gettext` lazy strings I should use either `StrPromise` or `StrOrPromise` (depending on the specific case) from `django-stubs-ext`.

Reply:
Sorry for not getting back to you earlier.

`StrOrPromise` accepts both `str` and `StrPromise`.

`StrPromise` only allows `StrPromse`, but not regular `str`.

In nearly all cases `StrOrPromise` is the correct one.